### PR TITLE
Angular 11 update

### PR DESCRIPTION
**Pull request by sdstolworthy**
Updates application to Angular 11. Also fixes typo that causes a failed unit test in angular > 9 (src/app/visits/visit-add/visit-add.component.spec.ts).

### CODE REVIEW

**Review by arey — COMMENTED** [review]
Thank you @sdstolworthy for your contribution.
I would like to know if you have followed the Angular Update Guide? https://update.angular.io/

File: package.json [hunk 1]
```diff
@@ -12,46 +12,47 @@
...
-    "@angular/platform-browser": "8.0.1",
-    "@angular/platform-browser-dynamic": "8.0.1",
-    "@angular/router": "8.0.1",
+    "@angular/animations": "11.2.11",
+    "@angular/cdk": "11.2.10",
+    "@angular/common": "11.2.11",
+    "@angular/compiler": "11.2.11",
+    "@angular/core": "11.2.11",
+    "@angular/forms": "11.2.11",
+    "@angular/material": "11.2.10",
```
Comment (arey) [inline package.json:21]: Why both material components remains in `11.2.10`?

File: package.json [hunk 2]
```diff
@@ -12,46 +12,47 @@
...
+    "zone.js": "~0.10.2"
   },
   "devDependencies": {
-    "@angular-devkit/build-angular": "~0.800.0",
-    "@angular/cli": "8.0.3",
-    "@angular/compiler-cli": "8.0.1",
-    "@angular/language-service": "8.0.1",
-    "@types/jasmine": "~2.8.3",
+    "@angular-devkit/build-angular": "~0.1102.10",
+    "@angular/cli": "11.2.10",
```
Comment (arey) [inline package.json:37]: Same question for Angular cli

**Comment by sdstolworthy** [comment]
Yes, I did. The application seemed to work as expected after the migration, and all the unit tests are passing.

File: src/app/app-routing.module.ts [hunk 3]
```diff
@@ -32,7 +32,7 @@ const appRoutes: Routes = [
 ];
 
 @NgModule({
-  imports: [RouterModule.forRoot(appRoutes)],
+  imports: [RouterModule.forRoot(appRoutes, { relativeLinkResolution: 'legacy' })],
```
Comment (arey) [inline src/app/app-routing.module.ts:35]: It is possible to migrate the application to the default value `corrected `?
https://angular.io/api/router/ExtraOptions

**Comment by arey** [comment]
Hi @sdstolworthy do you have time to solve minor issues I've notices? I really like to merge your PR.

**Comment by simrin051** [comment]
Hi @sdstolworthy / @arey 

Can i take up this issue

**Comment by arey** [comment]
It's ok for me. To finish the work of @sdstolworthy, there is some conflicts and comments to take into account.

**Comment by simrin051** [comment]
Sure, will work on it @arey

**Comment by simrin051** [comment]
Hi @arey 

When i ran this command as mentioned in  https://update.angular.io/  i get the error as in screenshot. Incompatibility in dependencies. Can you please advice.
 ng update @angular/core@11 @angular/cli@11 --allow-dirty


image

**Comment by simrin051** [comment]
Hi @arey 
When i ran the below command. I get error as in screenshot. Incompatibility in dependencies. Can you please advice.  
  ng update @angular/core@11 @angular/cli@11 --allow-dirty

image

**Comment by simrin051** [comment]
@arey , May i know the exact version to be upgraded (11.X.X).

**Pull request merged**